✨clusterctl: add support for the controlplane provider (#2099)
kubernetes-sigs/cluster-api: docs/book/src/clusterctl/commands/init.md

#### Automatically installed providers

The `clusterctl init` command automatically adds the `cluster-api` core provider, the `kubeadm-bootstrap` provider, and
the `kubeadm-controlplane` provider to the list of providers to install. This allows users to use a concise command syntax for initializing a management cluster. e.g.
use the command:

`clusterctl init --infrastracture aws`

> To get a fully operative management cluster with the `aws` infrastructure provider, the `cluster-api` provider, the `kubeadm-bootstrap` and the `kubeadm-controlplane` provider

Reviewer: ```suggestion
To get a fully operative management cluster with the `aws` infrastructure provider, the `cluster-api` core provider, the `kubeadm-bootstrap` and the `kubeadm-controlplane` provider
```

<aside class="note warning">

<h1> Warning </h1>

The `cluster-api` core provider, the `kubeadm-bootstrap` provider, and the `kubeadm-controlplane` provider are automatically installed only if:
- The user doesn't explicitly require to install a core/bootstrap/control-plane provider using the `--core` flag, the `--bootstrap` flag or the `--controlplane` flags;
- There is not an instance of a CoreProvider already installed in the cluster;

Please note that the second rule allows to execute `clusterctl init` more times: the first call actually initializes
the management cluster, while the subsequent calls can be used to add more providers.

</aside>

<aside class="note">

<h1> Is it possible to skip automatic install?</h1>

> In order to skip automatic provider installation set to `-` to the `--bootstrap` flag or to the `--controlplane` flags;

Reviewer 2: ```suggestion
In order to skip automatic provider installation use  `--bootstrap "-"` or  `--controlplane "-"`.
```

Reviewer: s/In order to/To/

😄

> instead it is not possible to skip automatic installation of the `cluster-api` core provider.

Reviewer 2: ```suggestion
Note it is not possible to skip automatic installation of the `cluster-api` core provider.
```
